src/CargoBuild/CargoBuildTask.cs

```diff
@@ -0,0 +1,437 @@
+        /// Gets or sets a feed name.
+        /// </summary>
+        public string RegistryFeedName { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets optional cargo command args.
+        /// </summary>
+        public string CommandArgs { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets a value indicating whether to use msrust up or not.
+        /// </summary>
+        public bool UseMsRustUp { get; set; } = false;
+
+        /// <inheritdoc/>
+        public override bool Execute()
+        {
+            // download & install rust if necessary
+            if (DownloadRustupAsync().GetAwaiter().GetResult())
+            {
+                if (InstallRust().GetAwaiter().GetResult())
+                {
+                    _shouldCleanRustPath = true;
+                }
+            }
+
+            if (Command.Equals("clearcargocache", StringComparison.InvariantCultureIgnoreCase))
+            {
+                if (Directory.Exists(_cargoHome))
+                {
+                    Log.LogMessage(MessageImportance.Normal, $"Clearing cargo cache at {_cargoHome}");
+                    Directory.Delete(_cargoHome, true);
+                }
+            }
+
+            if (!Command.Equals("fetch", StringComparison.InvariantCultureIgnoreCase))
+            {
+                var dir = Directory.GetParent(StartupProj!);
+                bool cargoFile = File.Exists(Path.Combine(dir!.FullName, "cargo.toml"));
+
+                return !cargoFile || CargoRunCommandAsync(Command.ToLower(), CommandArgs).GetAwaiter().GetResult() == ExitCode.Succeeded;
+            }
+
+            return FetchCratesAsync(StartupProj!).GetAwaiter().GetResult();
+        }
+
+        private async Task<ExitCode> CargoRunCommandAsync(string command, string args)
+        {
+            Log.LogMessage(MessageImportance.Normal, $"Executing cargo command: {command} {args}");
+            return await ExecuteProcessAsync(_cargoPath, $"{command} {args}", ".", _envVars);
+        }
+
+        private async Task<bool> FetchCratesAsync(string project)
+        {
+            var stopwatch = new Stopwatch();
+            try
+            {
+                stopwatch.Start();
+                Log.LogMessage(MessageImportance.Normal, $"---- CargoBuild fetch Starting ----\n\n");
+                var graphLoadStopWatch = new Stopwatch();
+                graphLoadStopWatch.Start();
+
+                var evaluationContext = EvaluationContext.Create(EvaluationContext.SharingPolicy.Shared);
+
+                var graph = new ProjectGraph(
+                    [new ProjectGraphEntryPoint(project)],
+                    ProjectCollection.GlobalProjectCollection,
+                    (string projectPath, Dictionary<string, string> globalProperties, ProjectCollection projCollection) =>
+                    {
+                        var loadSettings = ProjectLoadSettings.DoNotEvaluateElementsWithFalseCondition;
```

Comment:
Use `ProjectLoadSettings.IgnoreEmptyImports | ProjectLoadSettings.IgnoreInvalidImports | ProjectLoadSettings.IgnoreMissingImports | ProjectLoadSettings.DoNotEvaluateElementsWithFalseCondition`

Since this happens during the restore, Pkg variables will not be defined yet so imports of files inside packages will fail.

This uses the same flags that NuGet itself does during restore to load projects.